diff --git a/qt5/src/poppler-document.cc b/qt5/src/poppler-document.cc
index 68a73063..a27f302b 100644
--- a/qt5/src/poppler-document.cc
+++ b/qt5/src/poppler-document.cc
@@ -64,9 +64,11 @@ namespace Poppler {
   Document *Document::load(const QString &filePath, const QByteArray &ownerPassword,
 			   const QByteArray &userPassword)
     {
+    qDebug() << "<Poppler::Document::load(" << filePath << ")>";
 	DocumentData *doc = new DocumentData(filePath, 
 					     new GooString(ownerPassword.data()),
 					     new GooString(userPassword.data()));
+    qDebug() << "</Poppler::Document::load(" << filePath << ")>";
 	return DocumentData::checkDocument(doc);
     }
 
@@ -83,6 +85,7 @@ namespace Poppler {
     
     Document *DocumentData::checkDocument(DocumentData *doc)
     {
+    qDebug() << "<Poppler::DocumentData::checkDocument>";
 	Document *pdoc;
 	if (doc->doc->isOk() || doc->doc->getErrorCode() == errEncrypted) {
 		pdoc = new Document(doc);
@@ -97,8 +100,10 @@ namespace Poppler {
 	}
 	else
 	{
+		qDebug() << "Poppler::DocumentData::checkDocument - delete doc";
 		delete doc;
 	}
+    qDebug() << "</Poppler::DocumentData::checkDocument>";
 	return nullptr;
     }
 
diff --git a/qt5/src/poppler-private.cc b/qt5/src/poppler-private.cc
index 65311f85..a4ebe760 100644
--- a/qt5/src/poppler-private.cc
+++ b/qt5/src/poppler-private.cc
@@ -239,9 +239,12 @@ namespace Debug {
     
     DocumentData::~DocumentData()
     {
+	        qDebug() << "<Poppler::DocumentData::~DocumentData>";
         qDeleteAll(m_embeddedFiles);
         delete (OptContentModel *)m_optContentModel;
+	        qDebug() << "Poppler::DocumentData::~DocumentData before delete doc";
         delete doc;
+	        qDebug() << "Poppler::DocumentData::~DocumentData after delete doc";
     
         QMutexLocker locker{&mutex};
 
@@ -255,6 +258,7 @@ namespace Debug {
                 ownGlobalParams = false;
             }
         }
+	        qDebug() << "</Poppler::DocumentData::~DocumentData>";
       }
     
     void DocumentData::init()
diff --git a/qt5/src/poppler-private.h b/qt5/src/poppler-private.h
index 2ac843a2..a9281c0a 100644
--- a/qt5/src/poppler-private.h
+++ b/qt5/src/poppler-private.h
@@ -42,6 +42,7 @@
 #include <QtCore/QMutex>
 #include <QtCore/QPointer>
 #include <QtCore/QVector>
+#include <QtCore/QDebug>
 
 #include <config.h>
 #include <GfxState.h>
@@ -97,18 +98,23 @@ namespace Poppler {
     public:
 	DocumentData(const QString &filePath, GooString *ownerPassword, GooString *userPassword)
 	    {
+	        qDebug() << "<Poppler::DocumentData::DocumentData(" << filePath << ")>";
 		init();
+	        qDebug() << "Poppler::DocumentData::DocumentData after init";
 		m_filePath = filePath;	
 
 #ifdef _WIN32
 		doc = new PDFDoc((wchar_t *)filePath.utf16(), filePath.length(), ownerPassword, userPassword);
 #else
 		GooString *fileName = new GooString(QFile::encodeName(filePath).constData());
+	        qDebug() << "Poppler::DocumentData::DocumentData before new PDFDoc";
 		doc = new PDFDoc(fileName, ownerPassword, userPassword);
 #endif
+	        qDebug() << "Poppler::DocumentData::DocumentData after new PDFDoc";
 
 		delete ownerPassword;
 		delete userPassword;
+	        qDebug() << "</Poppler::DocumentData::DocumentData(" << filePath << ")>";
 	    }
 	
 	DocumentData(const QByteArray &data, GooString *ownerPassword, GooString *userPassword)
