diff --git a/qt5/src/poppler-document.cc b/qt5/src/poppler-document.cc
index bff01205..6261a325 100644
--- a/qt5/src/poppler-document.cc
+++ b/qt5/src/poppler-document.cc
@@ -61,9 +61,11 @@ namespace Poppler {
   Document *Document::load(const QString &filePath, const QByteArray &ownerPassword,
 			   const QByteArray &userPassword)
     {
+    qDebug() << "<Poppler::Document::load(" << filePath << ")>";
 	DocumentData *doc = new DocumentData(filePath, 
 					     new GooString(ownerPassword.data()),
 					     new GooString(userPassword.data()));
+    qDebug() << "</Poppler::Document::load(" << filePath << ")>";
 	return DocumentData::checkDocument(doc);
     }
 
@@ -89,6 +91,7 @@ namespace Poppler {
     
     Document *DocumentData::checkDocument(DocumentData *doc)
     {
+    qDebug() << "<Poppler::DocumentData::checkDocument>";
 	Document *pdoc;
 	if (doc->doc->isOk() || doc->doc->getErrorCode() == errEncrypted) {
 		pdoc = new Document(doc);
@@ -103,8 +106,10 @@ namespace Poppler {
 	}
 	else
 	{
+		qDebug() << "Poppler::DocumentData::checkDocument - delete doc";
 		delete doc;
 	}
+    qDebug() << "</Poppler::DocumentData::checkDocument>";
 	return nullptr;
     }
 
diff --git a/qt5/src/poppler-private.cc b/qt5/src/poppler-private.cc
index 81af2bed..d392d7af 100644
--- a/qt5/src/poppler-private.cc
+++ b/qt5/src/poppler-private.cc
@@ -232,9 +232,12 @@ namespace Debug {
     
     DocumentData::~DocumentData()
     {
+	        qDebug() << "<Poppler::DocumentData::~DocumentData>";
         qDeleteAll(m_embeddedFiles);
         delete (OptContentModel *)m_optContentModel;
+	        qDebug() << "Poppler::DocumentData::~DocumentData before delete doc";
         delete doc;
+	        qDebug() << "</Poppler::DocumentData::~DocumentData>";
       }
     
     void DocumentData::init()
diff --git a/qt5/src/poppler-private.h b/qt5/src/poppler-private.h
index 9131dba6..37b7d073 100644
--- a/qt5/src/poppler-private.h
+++ b/qt5/src/poppler-private.h
@@ -43,6 +43,7 @@
 #include <QtCore/QMutex>
 #include <QtCore/QPointer>
 #include <QtCore/QVector>
+#include <QtCore/QDebug>
 
 #include <config.h>
 #include <GfxState.h>
@@ -100,7 +101,10 @@ namespace Poppler {
 	DocumentData(const QString &filePath, GooString *ownerPassword, GooString *userPassword) :
 	GlobalParamsIniter(qt5ErrorFunction)
 	    {
+	        qDebug() << "<Poppler::DocumentData::DocumentData(" << filePath << ")>";
+
 		init();
+	        qDebug() << "Poppler::DocumentData::DocumentData after init";
 		m_device = nullptr;
 		m_filePath = filePath;	
 
@@ -108,11 +112,14 @@ namespace Poppler {
 		doc = new PDFDoc((wchar_t *)filePath.utf16(), filePath.length(), ownerPassword, userPassword);
 #else
 		GooString *fileName = new GooString(QFile::encodeName(filePath).constData());
+	        qDebug() << "Poppler::DocumentData::DocumentData before new PDFDoc";
 		doc = new PDFDoc(fileName, ownerPassword, userPassword);
 #endif
+	        qDebug() << "Poppler::DocumentData::DocumentData after new PDFDoc";
 
 		delete ownerPassword;
 		delete userPassword;
+	        qDebug() << "</Poppler::DocumentData::DocumentData(" << filePath << ")>";
 	    }
 	
 	DocumentData(QIODevice *device, GooString *ownerPassword, GooString *userPassword) :
