diff --git a/libtiff/tif_jpeg.c b/libtiff/tif_jpeg.c
index 250144f2..1c386c9c 100644
--- a/libtiff/tif_jpeg.c
+++ b/libtiff/tif_jpeg.c
@@ -107,8 +107,8 @@ int TIFFJPEGIsFullStripRequired_12(TIFF *tif);
 #ifndef __RPCNDR_H__ /* don't conflict if rpcndr.h already read */
 typedef unsigned char boolean;
 #endif
-#define HAVE_BOOLEAN /* prevent jmorecfg.h from redefining it */
 #endif
+#define HAVE_BOOLEAN /* prevent jmorecfg.h from redefining it */
 
 #include "jerror.h"
 #include "jpeglib.h"
