diff --git a/CMakeLists.txt b/CMakeLists.txt
index dbe6a089..80e5ec03 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -724,7 +724,6 @@ endif()
 set(PKG_CONFIG_VERSION_0_18 TRUE)
 if(PKG_CONFIG_EXECUTABLE)
   exec_program(${PKG_CONFIG_EXECUTABLE} ARGS --version RETURN_VALUE _return_VALUE OUTPUT_VARIABLE _output_VAR)
-  macro_ensure_version("0.18" "${_output_VAR}" PKG_CONFIG_VERSION_0_18)
 endif()
 if(PKG_CONFIG_VERSION_0_18)
   set(PC_REQUIRES "")
